commands/dictionary.js

```diff
@@ -0,0 +1,79 @@
+var GReYBot = require('../greybot');
+var util = require('util');
+var xml2js = require('xml2js');
+var parser = new xml2js.Parser();
+
+exports.commands = [
+	'define'
+];
+
+exports.define = {
+	usage: '<word>',
+	description: 'looks up a word in the Merriam-Webster Collegiate Dictionary',
+	process: (msg, suffix) => {
+		Array.prototype.clean = function(deleteValue) {
+			for (var i = 0; i < this.length; i++) {
+				if (this[i] == deleteValue) {
+				this.splice(i, 1);
+				i--;
+				}
+			}
+			return this;
+		};
+    	var word = suffix;
+		if (!word) {
+			msg.channel.send('I won\'t define an empty string.');
+			return;
+		}
+		var url = 'http://www.dictionaryapi.com/api/v1/references/collegiate/xml/' + word + '?key=' + GReYBot.Auth.dictionary_api_key;
+		require('request')(url,
+			function (err, res, body) {
+				let definitionResult = "";
+				parser.parseString(body, function (err, result) {
+					let wordList = result.entry_list.entry;
+					let phonetic = '';
+					phonetic += wordList[0].pr.map(entry => {
+						if (typeof(entry) === 'string') {
+							return entry;
+						}
+						if (entry['_']) {
+							return entry['_'];
+						}
+					}).join('\n');
+					definitionResult += '*['+phonetic+']*\n';
+					definitionResult += 'Hear it: http://media.merriam-webster.com/soundc11/'+wordList[0].sound[0].wav[0].slice(0,1)+'/'+wordList[0].sound[0].wav[0]+'\n\n';
+					wordList.forEach(wordResult => {
+						let defList = '';
+						let defArray = wordResult.def[0].dt;
+						if (wordResult.ew[0] !== word) { return; };
+						definitionResult += '__'+wordResult.fl[0]+'__\n';
+						defArray = defArray.filter(item => {
+							if (typeof(item) === 'string') {
+								return item !== ':' && item !== ': ';
+							}
+							if (item['_']) {
+								return item['_'] !== ':' && item['_'] !== ': ';
+							}
+						});
+						defList += defArray.map(entry => {
```

Comment:
Talked in chat. Determined to check it entry is an object and return _ value if so, otherwise just return entry so something is always returned.